### PR TITLE
Extra vehicle actions addon

### PR DESCRIPTION
**Pull request by micovery**
This is an addon for Arma 3 Wasteland that enables extra actions on vehicles.
- Lock/Unlock (from outside by vehicle owner)
- Lock/Unlock (from inside by anyone)
- Unflip vehicle
- Pull occupant (from outside by vehicle owner)
- Vehicle information

The addon is highly configurable (config.sqf), allowing server admins to
specify which actions they want enabled/disabled.

You could also extend this addon to put your own/extra vehicle actions, without needed to modify the A3Wasteland vanilla files.

### CODE REVIEW

File: addons/vactions/vector_functions.sqf [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    _matrix = [[cos(_angle), -(sin(_angle)), 0],
+               [sin(_angle), cos(_angle),    0],
+               [0,           0,              1]];
+  };
+  
+  ([_matrix, _v1] call matrix_X_vector)
+};
+
+vector_functions_defined = true;
+diag_log format["Loading vector functions complete"];
```
Comment (AgentRev) [inline addons/vactions/vector_functions.sqf:127]: Don't you know that there are already native commands for most of those operations?
Ctrl-F "vector": https://community.bistudio.com/wiki/Category:Scripting_Commands_Arma_3

vector_add = vectorAdd
vector_subtract = vectorDiff
vector_dotp = vectorDotProduct
vector_magnitude = vectorMagnitude
vector_resize = vectorMultiply
vector_angle = acos vectorCos

Comment (micovery) [inline addons/vactions/vector_functions.sqf:127]: Those are pretty new. Thanks for the hint. I will change them up. I wrote all this code back in Arma 2, when there was none of that. Most of this is porting.

**Comment by irepotato** [comment]
Hi,

For some reason I only get the lock option on vehicles I bought after a restart but the owner field is showing my player name before the restart, I just don't get the lock option

**Comment by Cael817** [comment]
there might be issues with 1.1 (vanilla) , i just tried to help another guy with
exactly the same issue.

Im pretty sure you dont get "ownerUID" set back to the vehicle after the
restart. (again with vanilla iniDB saving, socks works as it should)

you can test this, cut paste from my other mail =), only thing is that he
is using my old lockscripts but micoverys work on the same variables but
looking for car, motorcycle , tank plane etc but as the option works on
newly bought..... try the cursortarget thing below

---

My guess is that you dont get the "ownerUID" set back to the vehicle in
1.1, that should be the only thing the "action" looks for is if its a
landvehicle, air or ship. if the ownerUID matches the player and if its
locked or not. Try this, press U to get the admin menu, choose "access
proving grounds" choose "console" type in (or copy past from the mail)

cursortarget getvariable "ownerUID"

and execute, it should give a UID if set to the vehicle, im sure it is on
the newly bought as you have the unlock/lock option there so try on both a
saved and newly bought.
If its not showing anything on the saved one it either doesn't save (check
the vehicle save file for it) or it doesn't set it back even if its present
in the savefile.

2015-01-15 18:11 GMT+01:00 irepotato notifications@github.com:

**Comment by irepotato** [comment]
Hi,

Thanks for your reply, I try run that command and get this message 

cursortarget getvariable "ownerUID"
Error Missing ;

Thanks

**Comment by irepotato** [comment]
Oh just some info, with my server when you buy a new one you dont have the lock/unlock option, it is only after restart that the option shows

**Comment by micovery** [comment]
As Cael explained, the vanilla mission does not broadcast the "ownerUID" variable when you buy a vehicle, or when the vehicle is restored.

See this post http://forums.a3wasteland.com/index.php?topic=943.msg6062#msg6062 for explanation how to fix it in the vanilla code.

**Comment by irepotato** [comment]
Hi Micovery,

Thanks for the reply. Is there something I need to configure to change this? Before I upgraded to 1.1 it was working perfect. Or do you mean it doesn't broadcast with 1.1 vanilla? 

Thanks

**Comment by micovery** [comment]
By upgrading to v1.1 you probably undid whatever change you had made before that was broadcasting the "ownerUID" variable, so you have to re-do that change.

**Comment by LouDnl** [comment]
I've been getting complaints that the unflip script doesn't work correctly. Vehicles flip into the ground. Is it possible that you update the script using the commands Rev posted?

**Comment by LouDnl** [comment]
Here's a fix for the unflip function: https://github.com/LouDnl/ArmA3_Wasteland.Altis/commit/03328bc528c446313a06224a502f184fe44f0020

**Comment by lvlagic** [comment]
The vactions addon is broken in 1.54 update. You cannot now unlock vehicles, unflipping does not work either.

**Comment by codexGW** [comment]
The vactions addon is currently working with our mission on 1.54. Check out the diff and let us know if you find anything please.
https://github.com/A3Armory/ArmA3_Wasteland.Altis

**Comment by lvlagic** [comment]
The only differences i can see is the vehicle class is left blank for vehicles that can be locked/unlocked in the config.sqf meaning it should cover all, your one has vehicle classes listed there.

And in the va_functions.sqf at the end of the lines _action_id= you have got a different number at the end:
yours: [_player, _vehicle],0,false,false,"",

ours: [_player, _vehicle],-5,false,false,"",

I can try adding the class list when i get home tonight but if you can think of anything else i would be grateful to know. thanks.

**Comment by lvlagic** [comment]
OK, i re-adjusted the files to mirror the ones you have, and when i try to unlock a vehicle, i get the following error in the RPT log:

16:56:26 Scripting function 'a3w_fnc_lock' is not allowed to be remotely executed

I get a very similar error when trying to use the airdrop menu:

17:00:27 Scripting function 'apoc_srv_startairdrop' is not allowed to be remotely executed

Is this just a case of needing some sort of filter adding to the battleye remoteExec file or something?

**Comment by codexGW** [comment]
Maybe it has something to do with this function I added awhile back.
https://github.com/A3Armory/ArmA3_Wasteland.Altis/blob/dev/globalCompile.sqf#L73

**Pull request merged**